File: pascals-triangle-ii/pascals-triangle-ii.go
func getRow(rowIndex int) []int {
    leaf := make([]int, 0)
	for i := 0; i < rowIndex+1; i++ {
		prev := 1
		leaf = append(leaf, 1)
		for j := 1; i > 1 && j < i; j++ {
			curr := leaf[j]
			leaf[j] += prev
			prev = curr
		}
	}

	return leaf
}

File: pascals-triangle-ii/pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> leaf = new ArrayList<>();
        for (int i = 0; i < rowIndex+1; i++) {
            int prev = 1;
            leaf.add(1);
            for (int j = 1; i > 1 && j < i;j++) {
                int curr = leaf.get(j);
                leaf.set(j, prev + curr);
                prev = curr;
            }
        }

        return leaf;
    }
}